internal/subscription/model.go

```diff
@@ -25,7 +26,8 @@ type Model struct {
 	Labels                         map[string]string `json:"labels,omitempty"`
 	EnableMessageOrdering          bool              `json:"enable_message_ordering,omitempty"`
 	ExpirationPolicy               *ExpirationPolicy `json:"expiration_policy,omitempty"`
-	Filter                         string            `json:"filter,omitempty"`
+	FilterExpression               string            `json:"filter,omitempty"`
+	FilterStruct                   *Filter           `json:"filter_struct,omitempty"`
```

Comment:
do we need to save struct to db ? should we just save expression to db and have in memory struct 
can we just have that as a private attribute with getters ?

Reply:
Removed the struct from db model. Instead keeping it a private field with a Getter as suggested.